src/StaticLibrary.cpp

```diff
@@ -0,0 +1,193 @@
+#include "StaticLibrary.h"
+
+#include <stdio.h>
+#ifndef _MSC_VER
+#include <sys/stat.h>
+#include <unistd.h>
+#endif
+
+#include "Error.h"
+
+namespace Halide {
+namespace Internal {
+
+namespace {
+
+class File {
```

Comment:
Probably belongs in Util.h or similar so people don't reinvent it next time someone wants to do cross-platform file access.